app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,32 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    # Ensure all account endpoints are active/configured
+    account.switch do
+      cleanup_solr(account)
+      cleanup_fedora(account)
+      # Redis doesn't need 'account', as switch changes Redis::Namespace to correct namespace
+      cleanup_redis
```

Comment:
@atz : perhaps. Here's what I see in the code (if I'm understanding it correctly). 
* You are right, Solr doesn't require `switch`, as it passes in `connection_options` to Solr here: https://github.com/projecthydra-labs/hyku/blob/master/app/jobs/cleanup_account_job.rb#L22
* Fedora seems to require the `switch`. It had it previously (before my code refactor), and ActiveFedora seems to `register` a new `connection` via the switch call: https://github.com/projecthydra-labs/hyku/blob/master/app/models/fcrepo_endpoint.rb#L5  So, if I'm understanding `ActiveFedora.register` correctly, it seems like the connection is affected by `account.switch`
* Obviously, Redis requires `account.switch`

So, it looks like I could update this to only do `account.switch` for Fedora and Redis (and move the Solr cleanup outside that `do` & after Fedora & Redis are cleaned). Does that seem more reasonable to you, @atz?